src/ixdat/techniques/deconvolution.py

```diff
@@ -220,30 +153,146 @@ def fs(s):
+        TODO: add option to plot the implulse response from data/ return an axis to
+            co-plot with the data
+        TODO: figure out if it's ok to just get rid of dt and duration (not used here)
+        TODO: change the name kernel to something else
 
-        if norm:
-            area = np.trapz(kernel, t_kernel)
-            kernel = kernel / area
+        Args:
+            tspan (list): tspan over which to calculate the impulse response
+            tspan_bg (list): tspan of background to subtract. if list of tspans is
+                passed, will interpolate between the points using
+                calc_linear_background()
+            norm (bool): If true the impulse response is normalized to its
+                area. Default is True.
+            matrix (bool): If true the circulant matrix constructed from the
+                impulse reponse is returned. Default is False.
+        """
+        if self.type == "measured":
+            if type(tspan_bg[0]) is not list:
+                t_kernel, kernel = self.data.grab_flux(
+                    mol=self.mol, tspan=tspan, tspan_bg=tspan_bg
+                )
+            else:
+                t_kernel, kernel_raw = self.data.grab_flux(mol=self.mol, tspan=tspan)
+                bg = calc_linear_background(t_kernel, kernel_raw, tspans=tspan_bg)
+                kernel = kernel_raw - bg
+            if norm:
+                area = np.trapz(kernel, t_kernel)
+                kernel = kernel / area
+            if matrix:
+                kernel = np.tile(kernel, (len(kernel), 1))
+                i = 1
+                while i < len(t_kernel):  # TODO: pythonize this?
+                    kernel[i] = np.concatenate((kernel[0][i:], kernel[0][:i]))
+                    i = i + 1
+        else:
+            raise TechniqueError("Cannot calculate impulse response without data.")
+        return t_kernel, kernel
+
+    @np.vectorize
+    def get_cond_number(sampling_freq, working_dist):
+        """
+        Function to calculate condition number for a given sampling frequency
+        and working distance.
+        TODO: finish this method
+        """
+
+        # Define mass transport parameters.
+        params = {
+            "diff_const": 5.05e-9,
+            "work_dist": working_dist * 1e-6,
+            "vol_gas": 1.37e-10,
+            "volflow_cap": 1e-10,
+            "henry_vola": 52,
+        }
 
-        if matrix:
-            kernel = np.tile(kernel, (len(kernel), 1))
-            i = 1
-            while i < len(t_kernel):
-                kernel[i] = np.concatenate((kernel[0][i:], kernel[0][:i]))
-                i = i + 1
+        model_kernel = ECMSImpulseResponse(parameters=params)
+
+        kernel_matrix = model_kernel.calculate_kernel(
+            dt=1 / sampling_freq, duration=40, matrix=True
+        )
+
+        cond = np.linalg.cond(kernel_matrix)
+        print(str(cond))
+        return cond
+
+    def get_limit_frequency():
```

Comment:
These next three functions should likewise not be methods of the class if they don't take `self` as the first argument. They clearly won't work.

Could we just get rid of everything not in use? You can put in their place a comment with a permanent link to this code in this commit in case we need it in the future.